### ISSUE TITLE
ExecutePatchtAsync 

### ISSUE THREAD

**Issue by iwtu** [restsharp/RestSharp#1948]
Is not possible to call method PATCH wihout throwing an exception when request was not successful. 
Basically I'm missing the same funcionality as there is for GET, POST, PUT and DELETE methods.

**Comment by alexeyzimarev** [comment]
I bet you just need some overloads. Please feel free to open a PR to add them. These are basically copy-pasting from those you mentioned, like this:

```
    public static Task<RestResponse> ExecutePostAsync(this RestClient client, RestRequest request, CancellationToken cancellationToken = default)
        => client.ExecuteAsync(request, Method.Post, cancellationToken);
```

I would say that "impossible" is an overstatement as `ExecuteAsync(request, Method.Patch)` will do exactly what you want. Those extensions are just adding syntactic sugar over the base API.

**Comment by sebastienlabine** [comment]
Same goes for ExecuteDeleteAsync.

**Comment by alexeyzimarev** [comment]
I would greatly appreciate it if you guys submit PRs adding two LoC plus the XML comment for each of those missing extensions.

**Comment by akshay-zz** [comment]
@alexeyzimarev  Can I add them `ExecuteDeleteAsync `and `ExecutePatchtAsync`? I would be happy to contribute.

**Comment by iwtu** [comment]
To be honest, I would be happy to:

- remove Execute{METHOD}Async

OR 

- Execute{METHOD}Async will override client behaviour. 

It happened to me few times, that I used on request ExecutePostAsync but client for set to GET and it was really hard for me to notice that, because server behaved somehow correctly.

**Comment by alexeyzimarev** [comment]
> It happened to me few times

Could you elaborate? Do you mean you created a GET request and used `ExecutePostAsync`?

**Comment by alexeyzimarev** [comment]
> @alexeyzimarev Can I add them `ExecuteDeleteAsync `and `ExecutePatchtAsync`? I would be happy to contribute. Also is test cases required for this?

Of course

**Comment by iwtu** [comment]
> > It happened to me few times
> 
> Could you elaborate? Do you mean you created a GET request and used `ExecutePostAsync`?

yes. Just copy & paste * change only Execute. It happened to me pretty often.

**Comment by akshay-zz** [comment]
@alexeyzimarev  #1987  please review the PR.

**Comment by alexeyzimarev** [comment]
Fixed in #1996
